excise/utils.py

```diff
@@ -204,14 +196,19 @@ def append_line_to_data(
     ).first()
     warehouse_address = stock.warehouse.address if stock else None
 
-    unit_of_measure = variant.product.product_type.\
-        get_value_from_private_metadata(get_metadata_key("UnitOfMeasure"))
+    unit_of_measure = (
+        variant.product.product_type.get_value_from_private_metadata(
```

Comment:
Potential optimization issue. We should be sure that `product_type` is prefetched.

Reply:
What's do you suggest we should do for this and below instance as well?

Reply:
Not sure if this is a blocker for this PR, but we could make sure that the number of database queries is constant. If I'm correct, `append_line_to_data` is called in a for-loop somewhere in the plugin, and if the `product_type` is not prefetched, every time it is accessed, it will make a database query. To handle this we usually use `select_related` or `prefetch_related` on the queryset from which the `variant` instance comes from.

Reply:
For `append_line_to_data` called in `get_checkout_lines_data` we can pass to `append_line_to_data` `product_type` fetched from `CheckoutLineInfo`. 
For `append_line_to_data` called in `get_order_lines_data`, we should extend prefetch to fetch `product_type` also. Then we can pass `variant.product.product_type` directly to the `append_line_to_data`

Reply:
Here's the [PR (569ab01)](https://github.com/saleor/saleor-plugin-avatax-excise/pull/12/commits/569ab01)